### PR TITLE
add --proto to `goal clerk sign` and `goal clerk dryrun`

### PR DESCRIPTION
**Pull request by algobolson**

### CODE REVIEW

File: cmd/goal/clerk.go [hunk 1]
```diff
@@ -563,6 +564,22 @@ func lsigFromArgs(lsig *transactions.LogicSig) {
...
 }
 
+func getProto(versArg string) config.ConsensusParams {
+	cvers := protocol.ConsensusCurrentVersion
+	if protoVersion != "" {
+		cvers = protocol.ConsensusVersion(protoVersion)
+	}
+	proto, ok := config.Consensus[cvers]
+	if !ok {
+		fmt.Fprintf(os.Stderr, "invalid consensus version. possible versions:\n")
```
Comment (pzbitskiy) [inline cmd/goal/clerk.go]: Upper case Possible?

File: cmd/goal/clerk.go [hunk 2]
```diff
@@ -135,7 +136,7 @@ func init() {
 	dryrunCmd.Flags().StringVarP(&txFilename, "txfile", "t", "", "transaction or transaction-group to test")
 	dryrunCmd.Flags().Uint64VarP(&round, "round", "r", 1, "round number to simulate")
 	dryrunCmd.Flags().Int64VarP(&timeStamp, "time-stamp", "S", 0, "unix time stamp simulate (default now)")
-	dryrunCmd.Flags().StringVarP(&seedBase64, "seed", "R", "2JNZvkCE2KFv3wVbEIEsPPNR64ttpaWuD5NEIXqXc3g=", "base64 bytes to seed random number generator with")
+	dryrunCmd.Flags().StringVarP(&protoVersion, "proto", "P", "", "consensus protocol version id string")
```
Comment (pzbitskiy) [inline cmd/goal/clerk.go]: maybe put a default `config.Consensus[protocol.ConsensusCurrentVersion]` here and get rid of `getProto` function at all?

Comment (algobolson) [reply cmd/goal/clerk.go]: still gonna keep getProto for its error messaging

File: cmd/goal/clerk.go [hunk 3]
```diff
@@ -563,6 +564,22 @@ func lsigFromArgs(lsig *transactions.LogicSig) {
...
+	if protoVersion != "" {
+		cvers = protocol.ConsensusVersion(protoVersion)
+	}
+	proto, ok := config.Consensus[cvers]
+	if !ok {
+		fmt.Fprintf(os.Stderr, "Invalid consensus version. Possible versions:\n")
+		for xvers := range config.Consensus {
+			fmt.Fprintf(os.Stderr, "\t%s\n", xvers)
+		}
+		os.Exit(1)
```
Comment (tsachiherman) [inline cmd/goal/clerk.go]: nit: most of the goal code is using `reportErrorf` to print and exit.

Comment (algobolson) [reply cmd/goal/clerk.go]: yes, but it does one print and then exit, and I needed many prints and then an exit

**Pull request merged**